pkg/controller/daemon/update.go

```diff
@@ -153,13 +155,35 @@ func (dsc *DaemonSetsController) rollingUpdate(ctx context.Context, ds *apps.Dae
 			numSurge++
 			continue
 		}
+
+		// first count availability for all the nodes (even the ones that we are sunsetting due to scheduling constraints)
+		if oldPod != nil {
+			if podutil.IsPodAvailable(oldPod, ds.Spec.MinReadySeconds, metav1.Time{Time: now}) {
+				numAvailable++
+			}
+		} else if newPod != nil {
+			if podutil.IsPodAvailable(newPod, ds.Spec.MinReadySeconds, metav1.Time{Time: now}) {
+				numAvailable++
+			}
+		}
+
 		switch {
 		case oldPod == nil:
 			// we don't need to do anything to this node, the manage loop will handle it
 		case newPod == nil:
 			// this is a surge candidate
 			switch {
 			case !podutil.IsPodAvailable(oldPod, ds.Spec.MinReadySeconds, metav1.Time{Time: now}):
+				node, err := dsc.nodeLister.Get(nodeName)
+				if err != nil {
+					logger.Error(err, "Couldn't get node for nodeName", "nodeName", nodeName)
+					continue
```

Comment:
I think we should fail with error as before in these 3 instances, because we cannot produce a good decision when we cannot obtain a node object. The error would cause this to retry a moment later when for example a network situation would improve and we would get the node.

Reply:
Thanks, I'll let it fail as before.